stdlib/public/core/Codable.swift

```diff
@@ -2894,13 +2894,28 @@ public extension RawRepresentable where RawValue == String, Self : Decodable {
 //===----------------------------------------------------------------------===//
 
 // FIXME: Uncomment when conditional conformance is available.
-extension Array : Codable /* where Element : Codable */ {
-    public init(from decoder: Decoder) throws {
-        guard Element.self is Decodable.Type else {
-            preconditionFailure("Array<\(Element.self)> does not conform to Decodable because \(Element.self) does not conform to Decodable.")
+extension Array : Encodable /* where Element : Encodable */ {
```

Comment:
Sure! I think the primary source-breaking change will be that code which compiles now (but `preconditionFailure()`s) will fail to compile (which is what we'd like). What are you thinking of specifically in terms of generic algorithms though? Not sure I have the same thing in mind.